parsl/app/bash.py

```diff
@@ -60,15 +61,11 @@ def open_std_fd(fdname):
         stdfspec = kwargs.get(fdname)  # spec is str name or tuple (name, mode)
         if stdfspec is None:
             return None
-        elif isinstance(stdfspec, str):
-            fname = stdfspec
-            mode = 'a+'
-        elif isinstance(stdfspec, tuple):
-            if len(stdfspec) != 2:
-                raise pe.BadStdStreamFile("std descriptor %s has incorrect tuple length %s" % (fdname, len(stdfspec)), TypeError('Bad Tuple Length'))
-            fname, mode = stdfspec
-        else:
-            raise pe.BadStdStreamFile("std descriptor %s has unexpected type %s" % (fdname, str(type(stdfspec))), TypeError('Bad Tuple Type'))
+
+        try:
+            fname, mode = get_std_fname_mode(fdname, stdfspec)
+        except pe.BadStdStreamFile:
```

Comment:
does this `except` need to be here?

If get_std_fname_mode does not raise an exception, then this block doesn't fire.
If get_std_fname_mode raises a different exception, the exception propagates upwards.
If get_std_fname_mode raises a BadStdStreamFile exception, then this block propagates it upwards, which would be the behaviour anyway

Reply:
fixed